server/src/main/java/org/elasticsearch/indices/SystemIndexDescriptor.java

```diff
@@ -555,38 +555,19 @@ public MappingsVersion getMappingsVersion() {
      * @param cause the action being attempted that triggered the check. Used in the error message.
      * @return the standardized error message
      */
-    public String getMinimumMappingsVersionMessage(String cause) {
+    public String getMinimumMappingsVersionMessage(String cause, MappingsVersion requiredMinimumMappingVersion) {
         Objects.requireNonNull(cause);
         final MappingsVersion actualMinimumMappingsVersion = priorSystemIndexDescriptors.isEmpty()
             ? getMappingsVersion()
             : priorSystemIndexDescriptors.get(priorSystemIndexDescriptors.size() - 1).mappingsVersion;
         return Strings.format(
-            "[%s] failed - system index [%s] requires all data and master nodes to have mappings versions at least of version [%s]",
+            "[%s] failed - requested creation of system index [%s] with version [%s], which is greater than the cluster minimum index "
+                + "mapping version. Ensure that the system index descriptor for [%s] includes a prior definition for version [%s]",
```

Comment:
This message text seems confusing. As a user, I'd think if the requested version is _greater_ than the minimum, that should be acceptable.

Reply:
The message is really meant for developers, giving instruction on what to do.
It will read something like
"[auto-create] failed - requested creation of system index [.secrets] with version [2], which is greater than the cluster minimum index mapping version. Ensure that the system index descriptor for [.secrets] includes a prior definition for version [1]"

Maybe I can change it to 
"[auto-create] failed - requested creation of system index [.secrets] with version [2], which is greater than the   index mapping version supported by the cluster. Ensure that the system index descriptor for [.secrets] includes a prior definition for version [1]".

While it's true devs know that the cluster supports versions up to the minimum common of all versions from each node, maybe just saying "supported by the cluster" is better/clearer.

Reply:
@prdoyle Wdyt of the updated message?